proposals/0351-regex-builder.md

```diff
@@ -172,17 +169,25 @@ Regex builder addresses all of textual regexes' shortcomings presented in the [M
 
 ### `RegexComponent` protocol
 
-One of the goals of the regex builder DSL is allowing the developers to easily compose regexes from common currency types and literals, or even define custom patterns to use for matching. We introduce `RegexComponent`, a protocol that unifies all types that can represent a component of a regex.
+One of the goals of the regex builder DSL is allowing the developers to easily compose regexes from common currency types and literals, or even define custom patterns to use for matching. We introduce `RegexComponent` in the Swift Standard Library, a protocol that unifies all types that can represent a component of a regex. Since regexes are composable, the `Regex` type itself conforms to `RegexComponent`.
 
 ```swift
-public protocol RegexComponent {
-  associatedtype Output
-  @RegexComponentBuilder
+public protocol RegexComponent<RegexOutput> {
+  associatedtype RegexOutput
   var regex: Regex<Output> { get }
 }
+
+extension Regex: RegexComponent {
+  public typealias RegexOutput = Output
+  public var regex: Regex<Output> { self }
+}
 ```
 
-By conforming standard library types to `RegexComponent`, we allow them to be used inside the regex builder DSL as a match target.
+Note:
+- `RegexComponent` and `Regex`'s conformance to `RegexComponent` are available without importing `RegexBuilder`. All other types and conformances introduced in this proposal are in the `RegexBuilder` module.
+- The associated type `RegexOutput` intentionally has a `Regex` prefix. `Output` would cause confusion in standard library conforming types such as `String`, i.e. `String.Output`.
+
+By conforming standard library types to `RegexComponent`, we allow them to be used inside the regex builder DSL as a match target. These conformances are available in the `RegexBuilder` module.
```

Comment:
And also as arguments to the string processing algorithms (one reason to want the conformance in `Swift`).

Reply:
We resolve this: I had missed the fact the collection generic versions would apply instead of the regex ones. So it makes sense to keep the conformances in the regex builder module.